### PR TITLE
feat(wu-35): OTel trace pipeline — reset_telemetry API + seed_observability restructure

### PR DESCRIPTION
**Pull request by MacAttak**
## Summary

- Add `reset_telemetry()` public API to `floe_core.telemetry.initialization` for provider lifecycle management (shutdown, flag clear, set-once guard reset)
- Remove conflicting `otel_tracer_provider` fixture; restructure `seed_observability` to compile all 3 demo products with per-product `OTEL_SERVICE_NAME` using standard OTel env vars
- Update Jaeger service name expectations from generic `floe-platform` to product-specific `customer-360`

## Acceptance Criteria

| AC | Description | Status |
|----|-------------|--------|
| AC-35.1 | `reset_telemetry()` public API exists (importable, in `__all__`) | PASS — `initialization.py:129`, 15 unit tests |
| AC-35.2 | `otel_tracer_provider` fixture removed | PASS — 0 references in `tests/e2e/conftest.py` |
| AC-35.3 | `seed_observability` uses standard OTel env vars | PASS — `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_INSECURE` |
| AC-35.4 | All 3 demo products compiled with per-product service names | PASS — loop with `reset_telemetry()` between products |
| AC-35.5 | All 8 T2-affected tests pass | INFO — deferred to CI (requires Kind cluster) |
| AC-35.6 | Provider lifecycle properly managed | PASS — shutdown ordering, set-once guard reset, 6 lifecycle tests |

## Gate Results

| Gate | Status | Findings (B/W/I) |
|------|--------|-------------------|
| build | PASS | 0/0/0 |
| tests | PASS | 0/3/3 |
| security | PASS | 0/0/3 |
| wiring | PASS | 0/1/4 |
| spec | PASS | 0/0/1 |

## Key Decisions

- **OTel set-once guard reset**: `reset_telemetry()` resets `trace._TRACER_PROVIDER_SET_ONCE._done` (private API) because OTel Python has no public reset mechanism. Without this, `set_tracer_provider()` silently ignores subsequent calls. Pattern already used in 22+ test fixtures. Guarded with `hasattr()` for forward compat.
- **test_observability_roundtrip_e2e.py updated (unplanned)**: Consumed the removed `otel_tracer_provider` fixture — mechanical update to use global provider via `trace.get_tracer_provider()`.

## Test plan

- [x] 15 new unit tests for `reset_telemetry()` (import, shutdown, lifecycle, edge cases)
- [x] 317 existing telemetry unit tests pass (no regressions)
- [x] Lint clean (ruff), type-check clean (mypy --strict)
- [ ] E2E: 8 OTel/Jaeger tests pass with Kind cluster (CI)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/e2e/test_observability.py [hunk 1]
```diff
@@ -973,6 +974,18 @@ def test_compilation_emits_otel_spans(
...
+        from floe_core.telemetry.initialization import (
+            ensure_telemetry_initialized,
+            reset_telemetry,
+        )
+
+        os.environ["OTEL_EXPORTER_OTLP_ENDPOINT"] = "http://localhost:4317"
+        os.environ["OTEL_EXPORTER_OTLP_INSECURE"] = "true"
+        os.environ["OTEL_SERVICE_NAME"] = "customer-360"
+        reset_telemetry()
+        ensure_telemetry_initialized()
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply tests/e2e/test_observability.py]: Fixed in 8aabcc3. Test body now wraps OTel configuration in try/finally, saving/restoring all 3 env vars and calling `reset_telemetry()` in finally.

File: tests/e2e/test_observability_roundtrip_e2e.py [hunk 2]
```diff
@@ -161,26 +161,27 @@ def check_traces_in_jaeger() -> bool:
     @pytest.mark.requirement("AC-2.3")
     def test_otel_collector_accepts_spans(
         self,
-        otel_tracer_provider: Any,
+        seed_observability: None,
     ) -> None:
```
[the remark this answers is not in the document]
Comment (MacAttak) [reply tests/e2e/test_observability_roundtrip_e2e.py]: Acknowledged. The dependency on `seed_observability` (which requires Marquez) is intentional — it validates the real pipeline rather than a synthetic tracer. Marquez is already required for the E2E suite via `values-test.yaml`.
